src/components/AnonymousChat/child/ChatItemMyTextV2.tsx

```diff
@@ -115,55 +51,140 @@ const ChatItemMyTextV2 = ({
   message = '',
   status = ChatStatus.PENDING,
   avatar,
-  chatType
+  chatType,
+  messageType,
+  data
 }: ChatItemMyTextProps) => {
-  const messageRef = React.useRef<Text>(null);
+  const {
+    setReplyPreview,
+    onContextMenuPressed,
+    pulseAnimation,
+    animatedBubbleStyle,
+    animatedPulseStyle
+  } = useMessageHook();
+  const scrollContext = React.useContext(ScrollContext);
+
+  const swipeableRef = React.useRef<Swipeable>(null);
+  const [isNewLine, setIsNewLine] = React.useState(true);
+
+  React.useEffect(() => {
+    if (scrollContext?.selectedMessageId === data?.id) {
+      pulseAnimation.value = withSequence(
+        withDelay(300, withTiming(1.1, {duration: 200})),
+        withTiming(1, {duration: 200})
+      );
+      scrollContext?.setSelectedMessageId(null);
+    }
+  }, [scrollContext?.selectedMessageId]);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The use of hooks and context here is appropriate and well implemented. However, I noticed that you are not handling the case when `scrollContext` is undefined. This could potentially lead to a runtime error if `ScrollContext` does not provide a value. Consider adding a check for `scrollContext` before accessing its properties.

```diff
+  if (scrollContext) {
    React.useEffect(() => {
      if (scrollContext?.selectedMessageId === data?.id) {
        pulseAnimation.value = withSequence(
          withDelay(300, withTiming(1.1, {duration: 200})),
          withTiming(1, {duration: 200})
        );
        scrollContext?.setSelectedMessageId(null);
      }
    }, [scrollContext?.selectedMessageId]);
+  }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->